flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TestTaskBuilder.java

```diff
@@ -0,0 +1,240 @@
+		this.partitionProducerStateChecker = partitionProducerStateChecker;
+		return this;
+	}
+
+	public TestTaskBuilder setKvStateService(KvStateService kvStateService) {
+		this.kvStateService = kvStateService;
+		return this;
+	}
+
+	public TestTaskBuilder setExecutor(Executor executor) {
+		this.executor = executor;
+		return this;
+	}
+
+	public TestTaskBuilder setTaskManagerConfig(Configuration taskManagerConfig) {
+		this.taskManagerConfig = taskManagerConfig;
+		return this;
+	}
+
+	public TestTaskBuilder setExecutionConfig(ExecutionConfig executionConfig) {
+		this.executionConfig = executionConfig;
+		return this;
+	}
+
+	public TestTaskBuilder setRequiredJarFileBlobKeys(Collection<PermanentBlobKey> requiredJarFileBlobKeys) {
+		this.requiredJarFileBlobKeys = requiredJarFileBlobKeys;
+		return this;
+	}
+
+	public TestTaskBuilder setResultPartitions(Collection<ResultPartitionDeploymentDescriptor> resultPartitions) {
+		this.resultPartitions = resultPartitions;
+		return this;
+	}
+
+	public TestTaskBuilder setInputGates(Collection<InputGateDeploymentDescriptor> inputGates) {
+		this.inputGates = inputGates;
+		return this;
+	}
+
+	public Task build() throws Exception {
+		final JobID jobId = new JobID();
+		final JobVertexID jobVertexId = new JobVertexID();
+		final ExecutionAttemptID executionAttemptId = new ExecutionAttemptID();
+
+		final SerializedValue<ExecutionConfig> serializedExecutionConfig = new SerializedValue<>(executionConfig);
+
+		final JobInformation jobInformation = new JobInformation(
+			jobId,
+			"Test Job",
+			serializedExecutionConfig,
+			new Configuration(),
+			requiredJarFileBlobKeys,
+			Collections.emptyList());
+
+		final TaskInformation taskInformation = new TaskInformation(
+			jobVertexId,
+			"Test Task",
+			1,
+			1,
+			invokable.getName(),
+			new Configuration());
+
+		final BlobCacheService blobCacheService = new BlobCacheService(
+			mock(PermanentBlobCache.class),
+			mock(TransientBlobCache.class));
+
+		final TaskMetricGroup taskMetricGroup = UnregisteredMetricGroups.createUnregisteredTaskMetricGroup();
+
+		return new Task(
+			jobInformation,
+			taskInformation,
+			executionAttemptId,
+			new AllocationID(),
+			0,
+			0,
+			resultPartitions,
+			inputGates,
+			0,
+			mock(MemoryManager.class),
+			mock(IOManager.class),
+			shuffleEnvironment,
+			kvStateService,
+			mock(BroadcastVariableManager.class),
+			new TaskEventDispatcher(),
+			new TestTaskStateManager(),
+			taskManagerActions,
+			new MockInputSplitProvider(),
+			new TestCheckpointResponder(),
+			new TestGlobalAggregateManager(),
+			blobCacheService,
+			libraryCacheManager,
+			mock(FileCache.class),
+			new TestingTaskManagerRuntimeInfo(taskManagerConfig),
+			taskMetricGroup,
+			consumableNotifier,
+			partitionProducerStateChecker,
+			executor);
+	}
+
+	public static void setTaskState(Task task, ExecutionState state) {
+		try {
+			Field f = Task.class.getDeclaredField("executionState");
```

Comment:
Another option is that we make the method `Task#transitionState` as package private.